### PR TITLE
Port NixOS tests to python

### PR DESCRIPTION
**Pull request by kampka**
###### Motivation for this change

#72828

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @tfc

### CODE REVIEW

File: nixos/tests/paperless.nix [hunk 1]
```diff
@@ -13,17 +13,21 @@ import ./make-test.nix ({ lib, ... } : {
...
+    machine.succeed(
+        "convert -size 400x40 xc:white -font 'DejaVu-Sans' -pointsize 20 -fill black -annotate +5+20 'hello world 16-10-2005' /var/lib/paperless/consume/doc.png"
+    )
 
-    $machine->waitForUnit("paperless-server.service");
+    machine.wait_for_unit("paperless-server.service")
     # Wait until server accepts connections
-    $machine->waitUntilSucceeds("curl -s localhost:28981");
+    machine.wait_until_succeeds("curl -s localhost:28981")
     # Wait until document is consumed
```
Comment (tfc) [inline nixos/tests/paperless.nix]: these comments are good candidates for `subtest` titles.

File: nixos/tests/quagga.nix [hunk 2]
```diff
@@ -74,23 +74,23 @@ import ./make-test.nix ({ pkgs, ... }:
...
+          for gw in router1, router2:
+              gw.wait_for_unit("ospfd")
+              gw.wait_until_succeeds("vtysh -c 'show ip ospf neighbor' | grep Full")
+              gw.wait_until_succeeds("vtysh -c 'show ip route' | grep '^O>'")
 
           # Test ICMP.
-          $client->succeed("ping -c 3 server >&2");
+          client.succeed("ping -c 3 server >&2")
 
           # Test whether HTTP works.
```
Comment (tfc) [inline nixos/tests/quagga.nix]: these comments are also great candidates for `subtest` titles.

File: nixos/tests/paperless.nix (hunk 1 above)
Comment (kampka) [reply nixos/tests/paperless.nix]: Done

**Comment by veprbl** [comment]
@GrahamcOfBorg test systemd-timesyncd
@GrahamcOfBorg test switch-test
@GrahamcOfBorg test sonarr
@GrahamcOfBorg test quagga
@GrahamcOfBorg test postgis
@GrahamcOfBorg test php-pcre
@GrahamcOfBorg test pgmanage
@GrahamcOfBorg test peerflix
@GrahamcOfBorg test pdns-recursor
@GrahamcOfBorg test paperless

**Comment by veprbl** [comment]
@GrahamcOfBorg test switchTest

**Review by veprbl — APPROVED** [review]

**Comment by kampka** [comment]
@veprbl @tfc Thanks for reviewing :+1:

**Pull request merged**